test/conformance/enqueue/urEnqueueUSMMemcpy2D.cpp

```diff
@@ -25,15 +23,25 @@ struct urEnqueueUSMMemcpy2DTestWithParam
             GTEST_SKIP() << "2D USM memcpy is not supported";
         }
 
-        const auto [inPitch, inWidth, inHeight] = getParam();
-        std::tie(pitch, width, height) =
-            std::make_tuple(inPitch, inWidth, inHeight);
+        const auto [in2DParams, inSrcKind, inDstKind] = getParam();
+        std::tie(pitch, width, height, src_kind, dst_kind) =
+            std::make_tuple(in2DParams.pitch, in2DParams.width,
+                            in2DParams.height, inSrcKind, inDstKind);
+
+        ur_device_usm_access_capability_flags_t device_usm = 0;
+        ASSERT_SUCCESS(uur::GetDeviceUSMDeviceSupport(device, device_usm));
+        if (!device_usm && (src_kind == uur::USMKind::Device ||
+                            dst_kind == uur::USMKind::Device)) {
+            GTEST_SKIP() << "Device USM is not supported";
```

Comment:
we should move the memcpy2d support check from [here](https://github.com/oneapi-src/unified-runtime/pull/1233/files#diff-1b1b9f128a556f23bc1fbc8dbcabaa852e78b6d5b17cd2a76ef6994cd3be1d21R22) to after this one since failing that check implies we're going to fail this as well